File: python/refactoring/currency_converter.py
def convert_to(amount: float, to_currency: str) -> float:
    match to_currency:
        case "USD":
            return amount * 1.09
        case "EUR":
            return amount * 0.96
        case "GBP":
            return amount * 0.82
        case "RUB":
            return amount * 79.78
        case _:
            raise ValueError(f"no exchange rate for {to_currency} available")

File: csharp/Refactoring/CurrencyConverter.cs
namespace Refactoring;

public class CurrencyConverter
{
    private static Dictionary<string, double> exchangeRates = new()
        {
            { "USD", 1.09 },
            { "EUR", 0.96 },
            { "GBP", 0.82 },
            { "RUB", 79.78 }
        };

    public static double ConvertTo(double amount, string toCurrency)
    {
        if(exchangeRates.ContainsKey(toCurrency))
        {
            return amount * exchangeRates[toCurrency];
        }
        else
        {
            throw new ArgumentException($"no exchange rate for {toCurrency} available");
        }
    }
}
